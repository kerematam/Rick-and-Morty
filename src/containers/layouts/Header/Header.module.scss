.root {
  margin-top: 10px;
  margin-bottom: 10px;
}

.menu_item {
  display: inline-flex;
  .menu_item_button {
    margin-bottom: 15px;
  }
  .menu_item_icon {
    padding-right: 5px;
    padding-bottom: 3px;
  }
  .menu_item_text {
    text-transform: none;
  }
}
.menu_item_delimiter {
  margin-right: 3px;
  margin-top: 10px;
}
