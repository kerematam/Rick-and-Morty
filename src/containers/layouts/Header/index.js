import React from 'react'
import { useSelector } from 'react-redux'
import { Typography, Button } from '@material-ui/core/'
import {
  DoubleArrow as DoubleArrowIcon,
  PermIdentity as PermIdentityIcon,
  Home as HomeIcon,
} from '@material-ui/icons'
import history from 'utils/history'
import {
  characterDetailSelector,
  useCharacterDetailReduxInjector,
} from 'pages/CharacterPage/redux'

import styles from './Header.module.scss'

function handleClickHomepage() {
  history.replace('/characters')
}

export default function LayoutHeader() {
  useCharacterDetailReduxInjector()
  const pathname = useSelector(state => state.router.location.pathname)
  const { character } = useSelector(characterDetailSelector)
  const isCharacterDetailPage = !(
    pathname === '/characters' || pathname === '/characters/'
  )

  return (
    <div className={styles.root}>
      <span className={styles.menu_item}>
        <Button
          onClick={handleClickHomepage}
          className={styles.menu_iten_button}
        >
          <HomeIcon className={styles.menu_item_icon} fontSize="small" />
          <Typography className={styles.menu_item_text}>Characters</Typography>
        </Button>
      </span>
      {isCharacterDetailPage && (
        <>
          <span className={styles.menu_item_delimiter}>
            <DoubleArrowIcon />
          </span>
          <span className={styles.menu_item}>
            <Button className={styles.menu_iten_button}>
              <PermIdentityIcon
                className={styles.menu_item_icon}
                fontSize="small"
              />
              <Typography className={styles.menu_item_text}>
                {character.name}
              </Typography>
            </Button>
          </span>
        </>
      )}
    </div>
  )
}
