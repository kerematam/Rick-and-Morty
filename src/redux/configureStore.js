import { createStore, applyMiddleware, compose } from 'redux'
import createSagaMiddleware from 'redux-saga'
import createReducer from './rootReducers'
import { routerMiddleware } from 'connected-react-router'
import { createInjectorsEnhancer, forceReducerReload } from 'redux-injectors'

function configureStore(initialState = {}, history) {
  let composeEnhancers = compose
  const reduxSagaMonitorOptions = {}

  if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
  }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions)
  const { run: runSaga } = sagaMiddleware
  const middlewares = [sagaMiddleware, routerMiddleware(history)]

  const enhancers = [
    applyMiddleware(...middlewares),
    createInjectorsEnhancer({
      createReducer,
      runSaga,
    }),
  ]

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(...enhancers)
  )

  if (module.hot) {
    module.hot.accept('./rootReducers', () => {
      forceReducerReload(store)
    })
  }

  return store
}

export default configureStore
