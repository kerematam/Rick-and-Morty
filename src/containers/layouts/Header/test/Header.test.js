import React, { useCallback } from 'react'
import renderer from 'react-test-renderer'
import Header from '../index'
import { Provider } from 'react-redux'
import configureStore from 'redux/configureStore'

const setup = (setupProps = {}) => {
  const store = configureStore()
  const wrapper = renderer
    .create(
      <Provider store={store}>
        <Header {...setupProps} />
      </Provider>
    )
    .toJSON()

  return {
    wrapper,
  }
}

describe('Header', () => {
  test('should render correctly', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })
})
