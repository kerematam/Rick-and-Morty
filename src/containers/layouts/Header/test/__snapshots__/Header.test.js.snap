// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div
  className="root"
>
  <span
    className="menu_item"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text menu_item_button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root menu_item_icon MuiSvgIcon-fontSizeSmall"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
        </svg>
        <p
          className="MuiTypography-root menu_item_text MuiTypography-body1"
        >
          Characters
        </p>
      </span>
    </button>
  </span>
  <span
    className="menu_item_delimiter"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 5H11l5 7-5 7h4.5l5-7z"
      />
      <path
        d="M8.5 5H4l5 7-5 7h4.5l5-7z"
      />
    </svg>
  </span>
  <span
    className="menu_item"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text menu_iten_button"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root menu_item_icon MuiSvgIcon-fontSizeSmall"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
          />
        </svg>
        <p
          className="MuiTypography-root menu_item_text MuiTypography-body1"
        />
      </span>
    </button>
  </span>
</div>
`;
